### PR TITLE
Improvement: Reorganise and fix e2e tests

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: cypress/e2e/HomePage/list.project.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Feature: Test homepage: project list
+
+  Scenario: Open existing project
```
Comment (k-y0u) [inline cypress/e2e/HomePage/list.project.feature:3]: In this file, you should just check if the list of project is corresponding to the local storage.
AND
This test should be in another file named `open.existing.project.feature` to fit the other files name pattern.

File: src/components/ModelizerNavigationBar.vue [hunk 2]
```diff
@@ -20,7 +20,7 @@
       </router-link>
     </div>
     <div class="project-info">
-      <span class="project-name">{{ props.projectName }}</span>
+      <span class="project-name" data-cy="project-name">{{ props.projectName }}</span>
```
Comment (k-y0u) [inline src/components/ModelizerNavigationBar.vue:23]: Line should be divided.

File: cypress/support/step_definitions/context.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ Before(() => {
```
Comment (justinehell) [inline cypress/support/step_definitions/context.js:6]: imo this commit should be first since we use the new step added in other files

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 4]
```diff
@@ -0,0 +1,151 @@
+Feature: Test git configuration dialog
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:1]: redundancy of `settings` in file name (second not necessary)
AND
file name -> settings != test name -> configuration (need to make a choice)

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    Given I clear cache
+    And I set viewport size to "1536" px for width and "960" px for height
+    And I visit the "/"
+
+    When I click on "[data-cy=\"new-project\"]"
+    Then I expect current url is "/modelizer/project-[a-f0-9]{8}/model"
+    And  I extract "project-[a-f0-9]{8}" from url in field "projectName" of context
+    And I visit the "/#/modelizer/{{ projectName }}/text"
+
+  Scenario: Set git configuration for a project
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:13]: `Set git configuration for a project` -> `Set correct git configuration in the project`

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  Scenario: Set empty token should display an error
+    When I click on "[data-cy=\"project-settings\"]"
+    And I click on "[data-cy=\"git-settings-menu\"]"
+    And I set on "[data-cy=\"git-form\"] [data-cy=\"git-repository-input\"]" text "https://github.com/ditrit/leto-modelizer-project-test"
+    And I set on "[data-cy=\"git-form\"] [data-cy=\"git-username-input\"]" text "test"
+    And I click on "[data-cy=\"git-form\"] [data-cy=\"git-form-submit\"]"
+    Then I expect field "[data-cy=\"git-form\"] [data-cy=\"git-token-input\"]" is ""
+    And I expect "[data-cy=\"git-form\"] [role=\"alert\"]" is "Please type something"
+
+  Scenario: Each project should have its own value of git configuration
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:58]: I don't understand the purpose of this test.

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+    Then I expect "positive" toast to appear with text "We have access to your repository 🥳!"
+    And I expect "[data-cy=\"git-form\"]" is closed
+
+    When I click on "[data-cy=\"git-current-branch\"]"
+    Then I expect "[data-cy=\"git-menu-branch-local-master\"]" exists
+    And I expect "[data-cy=\"git-menu-branch-local-master\"]" appear 1 time on screen
+
+
+  Scenario: Save git configuration twice should not duplicate branches in branch menu
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:106]: duplication

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    And I click on "[data-cy=\"git-form\"] [data-cy=\"git-form-submit\"]"
+
+    Then I expect "positive" toast to appear with text "We have access to your repository 🥳!"
+    And I expect "[data-cy=\"git-form\"]" is closed
+
+    When I click on "[data-cy=\"git-current-branch\"]"
+    Then I expect "[data-cy=\"git-menu-branch-local-master\"]" exists
+    And I expect "[data-cy=\"git-menu-branch-local-master\"]" appear 1 time on screen
+
+  Scenario: Expand menu should be closed when menu is open
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:132]: should be put in another test file cause only concern TextView. (AND not even git setting/config)

File: cypress/e2e/ModelizerPage/TextView/file/open.file.feature [hunk 9]
```diff
@@ -1,32 +1,23 @@
-Feature: Test file explorer and file tabs
+Feature: Test modelizer text view: open file/folder
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/TextView/file/open.file.feature:1]: can't open a folder, only expand. BUT can't expand a file either. It's only possible to double click on both but the result won't be the same.
AND
Does not match file name.
AND
Folder text missing.

File: cypress/e2e/ModelizerPage/TextView/git/display.branch.feature [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+    When I click on "[data-cy=\"new-project\"]"
+    Then I expect current url is "/modelizer/project-[a-f0-9]{8}/model"
+    And  I extract "project-[a-f0-9]{8}" from url in field "projectName" of context
+    And I visit the "/#/modelizer/{{projectName}}/text"
+
+  Scenario: Default branch should be master
+    Then I expect "[data-cy=\"git-current-branch\"] " is "master"
+
+  Scenario: Expect to have master in local branches section
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/TextView/git/display.branch.feature:16]: what about other branches than locales ?

File: cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  Scenario: Expect to have no branch with "no_main" filter
+    When I click on "[data-cy=\"git-current-branch\"]"
+    Then I expect "[data-cy=\"git-menu-branch-local-main\"]" exists
+    And I expect "[data-cy=\"git-menu-branch-remote-main\"]" exists
+
+    When I set on "[data-cy=\"git-branch-menu-search-bar\"]" text "not_main"
+    Then I expect "[data-cy=\"git-menu-branch-local-main\"]" not exists
+    And I expect "[data-cy=\"git-menu-branch-remote-main\"]" not exists
+
+  Scenario: Expect to have main branch with "main" filter
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature:27]: Hard to know if the test works if result does not change.

File: cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    When I click on "[data-cy=\"git-current-branch\"]"
+    Then I expect "[data-cy=\"git-menu-branch-local-main\"]" exists
+    And I expect "[data-cy=\"git-menu-branch-remote-main\"]" exists
+
+    When I set on "[data-cy=\"git-branch-menu-search-bar\"]" text "main"
+    Then I expect "[data-cy=\"git-menu-branch-local-main\"]" exists
+    And I expect "[data-cy=\"git-menu-branch-remote-main\"]" exists
+
+  Scenario: Expect to have main branch with "no_main main" filter
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature:36]: Hard to know if the test works if result does not change.

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/components/ModelizerNavigationBar.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/ModelizerNavigationBar.vue:23]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+    And I click on "[data-cy=\"git-settings-menu\"]"
+    And I click on "[data-cy=\"git-form\"] [data-cy=\"git-form-submit\"]"
+
+    Then I expect "positive" toast to appear with text "We have access to your repository 🥳!"
+    And I expect "[data-cy=\"git-form\"]" is closed
+
+    When I click on "[data-cy=\"git-current-branch\"]"
+    Then I expect "[data-cy=\"git-menu-branch-local-master\"]" exists
+    And I expect "[data-cy=\"git-menu-branch-local-master\"]" appear 1 time on screen
+
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:131]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+    Then I expect current url is "/"
+
+    When I click on "[data-cy=\"new-project\"]"
+    And I click on "[data-cy=\"project-settings\"]"
+    And I click on "[data-cy=\"git-settings-menu\"]"
+    Then I expect field "[data-cy=\"git-form\"] [data-cy=\"git-repository-input\"]" is ""
+    And I expect field "[data-cy=\"git-form\"] [data-cy=\"git-username-input\"]" is ""
+    And I expect field "[data-cy=\"git-form\"] [data-cy=\"git-token-input\"]" is ""
+
+
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:78]: done

File: cypress/e2e/ModelizerPage/AllView/switch.mode.feature [hunk 15]
```diff
@@ -1,11 +1,11 @@
-Feature: Test modelizer page
+Feature: Test modelizer page: switch mode (text/model)
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/switch.mode.feature:1]: done

File: cypress/e2e/ModelizerPage/AllView/switch.mode.feature [hunk 16]
```diff
@@ -1,11 +1,11 @@
...
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/switch.mode.feature:2]: done

File: cypress/e2e/HomePage/create.project.feature [hunk 17]
```diff
@@ -0,0 +1,17 @@
+Feature: Test homepage: project creation
+
+  Scenario: Create project should redirect to project page
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/HomePage/create.project.feature:3]: done

File: cypress/e2e/HomePage/create.project.feature [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+    And I visit the "/"
+
+    When I click on "[data-cy=\"new-project\"]"
+    Then I expect current url is "modelizer/project-[a-f0-9]{8}/model"
+
+  Scenario: Create project should save it in browser cache
+    Given I clear cache
+    And I visit the "/"
+
+    When I click on "[data-cy=\"new-project\"]"
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/HomePage/create.project.feature:14]: done

File: cypress/e2e/HomePage/import.project.feature [hunk 19]
```diff
@@ -0,0 +1,17 @@
+Feature: Test homepage: project import
+
+  Scenario: Create project should redirect to project page
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/HomePage/import.project.feature:3]: done

File: src/components/ModelizerNavigationBar.vue (hunk 2 above)
Comment (Zorin95670) [reply src/components/ModelizerNavigationBar.vue:23]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 4 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:1]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 5 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:13]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 6 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:58]: If I save configuration, it won't update all other configuration, check that configuration is related to a project

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 7 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:106]: vone

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 8 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature:132]: done

File: cypress/e2e/ModelizerPage/TextView/git/display.branch.feature (hunk 10 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/TextView/git/display.branch.feature:16]: we don't have other branch on git init

File: cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature (hunk 12 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature:36]: yeah but the test work !

File: cypress/e2e/HomePage/create.project.feature (hunk 17 above)
[the remark this answers is not in the document]
Comment (k-y0u) [reply cypress/e2e/HomePage/create.project.feature]: my bad : `to project modelizer page` -> `to project model view page`

File: cypress/e2e/HomePage/create.project.feature (hunk 18 above)
[the remark this answers is not in the document]
Comment (k-y0u) [reply cypress/e2e/HomePage/create.project.feature:12]: missing the third step, why ?

File: cypress/e2e/HomePage/list.project.feature (hunk 1 above)
Comment (k-y0u) [reply cypress/e2e/HomePage/list.project.feature]: no file anymore... where is the test to `check if the list of project is corresponding to the local storage` ?

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 4 above)
Comment (k-y0u) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature]: second point still present

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 6 above)
Comment (k-y0u) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature]: I just deeply check and I understand that your test :
* create a new project
* set la git config
* back to homepage
* create a new project
-> the new project has no git config
So this test should check if the config of the first project on local storage is (still) existing to be accurate.
(could be the set of git config which didn't work...)

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 8 above)
Comment (k-y0u) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature]: should be in TextView tests...

File: cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature (hunk 12 above)
Comment (k-y0u) [reply cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature:40]: nope

File: cypress/e2e/ModelizerPage/TextView/git/common.action.feature [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+  Background:
+    Given I clear cache
+    And I visit the "/"
+
+    When I click on "[data-cy=\"new-project\"]"
+    Then I expect current url is "/modelizer/project-[a-f0-9]{8}/model"
+    And  I extract "project-[a-f0-9]{8}" from url in field "projectName" of context
+    And I visit the "/#/modelizer/{{ projectName }}/text"
+
+  Scenario: Expand menu should be closed when menu is open
```
Comment (justinehell) [inline cypress/e2e/ModelizerPage/TextView/git/common.action.feature:12]: I don't feel the text fits what's done in this scenario ?

File: tests/unit/composables/Project.spec.js [hunk 21]
```diff
@@ -237,9 +237,11 @@ describe('Test composable: Project', () => {
   });
 
   describe('Test function: initProject', () => {
-    it('should call git init', async () => {
-      const result = await initProject({ id: 'foo' });
-      expect(result).toEqual('init');
+    it('should init create commit with default file', async () => {
```
Comment (justinehell) [inline tests/unit/composables/Project.spec.js:240]: ` init create commit` -> `init and create commit` ?

File: cypress/support/step_definitions/html.js [hunk 22]
```diff
@@ -41,7 +41,7 @@ Then('I set on {string} text {string}', (templateSelector, templateValue) => {
   const selector = nunjucks.renderString(templateSelector, cy.context);
   const value = nunjucks.renderString(templateValue, cy.context);
 
-  cy.get(selector).type(value);
+  cy.get(selector).clear().type(value);
```
Comment (justinehell) [inline cypress/support/step_definitions/html.js:44]: why `.clear()` ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/support/step_definitions/html.js (hunk 22 above)
Comment (Zorin95670) [reply cypress/support/step_definitions/html.js:44]: to clear input before typing text, make a functionnal test bug with the branch filter

File: tests/unit/composables/Project.spec.js (hunk 21 above)
Comment (Zorin95670) [reply tests/unit/composables/Project.spec.js:240]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 4 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature]: done

File: cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature (hunk 8 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/AllView/settings/git.settings.feature]: done

File: cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature (hunk 11 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature:27]: the test is change, done

File: cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature (hunk 12 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/TextView/git/filter.branch.feature:40]: the test is change done

File: src/components/ModelizerNavigationBar.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/ModelizerNavigationBar.vue]: redo ?

**Review by justinehell — CHANGES_REQUESTED** [review]

**Review by justinehell — APPROVED** [review]

**Review by justinehell — DISMISSED** [review]

File: cypress/e2e/HomePage/create.project.feature (hunk 17 above)
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/HomePage/create.project.feature]: done

File: cypress/e2e/HomePage/import.project.feature (hunk 19 above)
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/HomePage/import.project.feature]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**